<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>HiC - Metagenome tutorial</title>
        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../css/highlight.css">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->
	
	<script src="../js/jquery-1.10.2.min.js"></script>
        <script src="../js/bootstrap-3.0.3.min.js"></script>
        <script src="../js/highlight.pack.js"></script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="..">Metagenome tutorial</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                    <li >
                        <a href="..">Home</a>
                    </li>
                    <li >
                        <a href="../startup/">Jupyter</a>
                    </li>
                    <li >
                        <a href="../qc/">FastQC</a>
                    </li>
                    <li >
                        <a href="../assembly/">Assembly</a>
                    </li>
                    <li >
                        <a href="../binning/">Binning</a>
                    </li>
                    <li class="active">
                        <a href="./">HiC</a>
                    </li>
                </ul>

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                    <li >
                        <a rel="next" href="../binning/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li class="disabled">
                        <a rel="prev" >
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
            </ul>
        </div>
    </div>
</div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#metagenomic-hic">Metagenomic HiC</a></li>
            <li><a href="#what-is-metagenomic-hic">What is Metagenomic HiC?</a></li>
            <li><a href="#the-hi-c-library-protocol">The Hi-C library protocol</a></li>
            <li><a href="#protocol-outline">Protocol outline</a></li>
        <li class="main "><a href="#qc-is-my-library-ok">QC: is my library ok?</a></li>
        <li class="main "><a href="#metagenome-binning-with-hic-data">Metagenome binning with HiC data</a></li>
        <li class="main "><a href="#references">References</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="metagenomic-hic">Metagenomic HiC</h1>
<h2 id="what-is-metagenomic-hic">What is Metagenomic HiC?</h2>
<p>As the name suggest, Metagenomic Hi-C incorporates a new form of sequencing datas, named Hi-C, in to traditional shotgun metagenomes. In doing so, modern analyses, such as genome binning, can be performed accurately and precisely using only a single time-point.</p>
<p>The Hi-C protocol was designed to capture genome-wide evidence of DNA molecules in close physical proximity <em>in vivo</em>, prior to cellular lysis. With the flexible nature of a chromosome and its capacity to bend back on itself, the most frequently captured interactions are usually those between loci on the same chromosome. After this, the most frequently observed interactions are inter-chromosomal. Lastly come inter-cellular interactions, which are often far below the rates of those found within a cell.</p>
<p>The original purpose of Hi-C was to study the 3-dimensional conformation of the human genome [1], but the proximity information can just as well be exploited to infer which DNA assembly fragments belong together in a bin.  Depending on the subject of study, the implied precision of "together" can range from a genome to chromosome. For metagenomic studies, methods currently aim to associate assembly fragments believed to come from the same genome.</p>
<h2 id="the-hi-c-library-protocol">The Hi-C library protocol</h2>
<p>Hi-C data-sets are generated using conventional high-throughput Illumina paired-end sequencing; the difference lies in the library protocol.</p>
<h2 id="_1"></h2>
<blockquote>
<p>:camera: <strong>Fig 1</strong> Major steps of the Hi-C protocol
<img alt="zoomify" src="../img/hi-c_steps.png" />{.center .small}</p>
</blockquote>
<h2 id="_2"></h2>
<h3 id="protocol-outline">Protocol outline</h3>
<ol>
<li>
<p><strong>DNA Fixation:</strong> Beginning with intact cells, the first step of the Hi-C protocol is formalin fixation. The act of cross-linking freezes close-by conformation arrangements within the DNA that existed in the cells at the time of fixation.</p>
</li>
<li>
<p><strong>Cell Lysis:</strong> The cells are lysed and DNA-protein complexes extracted and purified.</p>
</li>
<li>
<p><strong>Restriction Digest:</strong> The DNA is digested using a restriction endonuclease. In metagenomic Hi-C, enzymes with large overhangs and 4-nt recognition sites are common choices (i.e. Sau3AI, MluCI, DpnII). Differences in the GC bias of the recognition site and the target DNA is an important factor, as inefficient digestion will produce fewer Hi-C read-pairs.</p>
</li>
<li>
<p><strong>Biotin Tagging:</strong> The overhangs produced during digestion are end-filled with biotinylated nucleotides.</p>
</li>
<li>
<p><strong>Free-end Ligation:</strong> in dilute conditions or immobilised on a substrate, free-ends protruding from the DNA-protein complexes are ligated. This stochastic process favours any two ends which were nearby within the complex, however random ligation and self-ligation can occur and are non-informative.</p>
</li>
<li>
<p><strong>Crosslink Reversal:</strong> The formalin fixation is reversed, allowing the now free DNA to be purified.</p>
</li>
<li>
<p><strong>Un-ligated End Clean-up:</strong> Free-ends which failed to form ligation products are unwanted in subsequent steps but could still be biotin tagged. To minimise their inclusion, a light exonuclease 3' to 5' favoured chew-back can be applied.</p>
</li>
<li>
<p><strong>DNA Sheering:</strong> With ligation completed, the DNA is mechanically sheered and the size range of the resulting fragment selected suitability with Illumina paired-end sequencing. </p>
</li>
<li>
<p><strong>DNA repair:</strong> Sonication can lead to damaged ends and nicks, which can be repaired.</p>
</li>
<li>
<p><strong>Proximity Ligation Enrichment:</strong> Biotin tagged fragments are pulled down using affinity purification. </p>
</li>
<li>
<p><strong>Adpater Ligation:</strong> Illumina paired-end adapter ligation and associated steps to produce a sequencing library are now applied.</p>
</li>
</ol>
<h2 id="qc-is-my-library-ok">QC: is my library ok?</h2>
<h2 id="metagenome-binning-with-hic-data">Metagenome binning with HiC data</h2>
<h2 id="references">References</h2>
<ol>
<li>Lieberman-Aiden, E., van Berkum, N. L., Williams, L., Imakaev, M., Ragoczy, T., Telling, A., … Dekker, J. (2009). Comprehensive mapping of long-range interactions reveals folding principles of the human genome. <em>Science</em>, 326(5950), 289–293. https://doi.org/10.1126/science.1181369</li>
</ol></div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>var base_url = '..';</script>
        <script data-main="../mkdocs/js/search.js" src="../mkdocs/js/require.js"></script>
        <script src="../js/base.js"></script><div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
